### PR TITLE
SOAR-8575 Separated UseCaseValidator for workflow and plugins validation

### PR DESCRIPTION
**Pull request by tcheng-r7**
### Description

Describe the proposed changes:

  - Currently the UseCaseValidator is used for both plugin and workflow validation. With the large update to workflow use case tags, workflows can no longer share the same use cases with plugins. The new WorkflowUseCaseValidator pulls in use cases from the Hub API and checks PRs against those use cases. 

## PR Requirements

Developers, verify you have completed the following items by checking them off:

- [x] Unit tests written for any new or updated code
- [x] Version bumped within setup.py
- [x] Changelog entry added

### CODE REVIEW

**Review by mberezin-r7 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ be on your way to contributing!
 
 ## Changelog
 
+* 2.44.3 - Separated UseCaseValidator for workflows as plugins and workflows now have different usecase tags
```
Comment (mrinehart-r7) [inline README.md:65]:
```suggestion
* 2.45.0- Separated UseCaseValidator for workflows as plugins and workflows now have different usecase tags
```
Minor version bump as it's technically a new validator

File: icon_validator/rules/workflow_validators/workflow_use_case_validator.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        approved_usecases = []
+        query = "types=usecase"
+        for i in range(0, 9999):
+            response = requests.get(url=f"https://extensions-api.rapid7.com/v2/public/tags?first=1000&{query}")
+            response_json = response.json()
+            approved_usecases.extend(WorkflowUseCaseValidator.get_approved_usecase_tags(response_json))
+            if not response_json["pageInfo"]["hasNextPage"]:
+                break
+
+            query = f"after={response_json['pageInfo']['endCursor']}"
```
Comment (mrinehart-r7) [inline icon_validator/rules/workflow_validators/workflow_use_case_validator.py:29]:
```suggestion
        params = {
            "first": 1000
            "types": "usecase"
        }
        
        has_next = True
        while has_next:
            response = requests.get(url=f"https://extensions-api.rapid7.com/v2/public/tags", params=params)
            response_json = response.json()
            approved_usecases.extend(WorkflowUseCaseValidator.get_approved_usecase_tags(response_json))
            if not response_json["pageInfo"]["hasNextPage"]:
                has_next = False

            params["after"] = response_json['pageInfo']['endCursor']
```
Cleaned up the request a bit - you can pass dictionaries into the `params` arg in requests.get() to use query parameters in a cleaner way. Also switched the for loop to a while. I highly doubt we'd ever paginate 10000 times, but you never know 🤷🏻 . Also fixed the query update on line 29 - you were clearing out the existing params which was yielding an HTTP 400 when I tried it in the browser.

Give this code a shot

File: icon_validator/rules/workflow_validators/workflow_use_case_validator.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+        return approved_usecases
+
+    @staticmethod
+    def validate_use_case_exists(spec: KomandPluginSpec):
+        try:
+            use_cases = spec.spec_dictionary()["hub_tags"]["use_cases"]
+            if not use_cases or not len(use_cases):
```
Comment (mrinehart-r7) [inline icon_validator/rules/workflow_validators/workflow_use_case_validator.py:37]:
```suggestion
            if not use_cases:
```
You can consolidate them as `if not var` checks for False, None, and 0 length

File: icon_validator/rules/workflow_validators/workflow_use_case_validator.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    def get_approved_usecase_tags(response_json: dict) -> [str]:
+        approved_usecases = []
+        for result in response_json["results"]:
+            if result["type"] == "usecase":
+                approved_usecases.append(result["name"])
+
+        return approved_usecases
+
+    @staticmethod
+    def get_approved_usecase_tags_with_paging() -> [str]:
```
Comment (mrinehart-r7) [inline icon_validator/rules/workflow_validators/workflow_use_case_validator.py:19]: If you want a fun thing to try (not required for this PR, but could improve performance if this usecase list ever grew huge), you can turn this function into a [generator](https://wiki.python.org/moin/Generators) that yields usecases that can be compared against. You could then try comparing all the usecases in the workflow against the current set, and if they don't match, then get the next set from the generator (which would be the next page from the API), and continue until all possible usecase matches have been exhausted.

File: setup.py [hunk 5]
```diff
@@ -7,7 +7,7 @@
 
 setup(
     name="insightconnect_integrations_validators",
-    version="2.44.2",
+    version="2.44.3",
```
Comment (mrinehart-r7) [inline setup.py:10]:
```suggestion
    version="2.45.0",
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by mberezin-r7 — APPROVED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Pull request merged**